src/utils.js

```diff
@@ -41,4 +41,47 @@ const get = (url, responseType) => {
   });
 };
 
-export { get };
+/*
+  returns multiple json in the event that we are dealing with a multi-url query
+*/
+const multiGet = (urlList, responseType) => {
+  let promises = [];
+  urlList.forEach(url => {
+    let newPromise = new Promise(function(resolve, reject) {
+      // Do the usual XHR stuff
+      let req = new XMLHttpRequest();
+      req.responseType = responseType;
+
+      req.open("GET", url);
+
+      req.onload = function() {
+        // This is called even on 404 etc
+        // so check the status
+        if (req.status == 200) {
+          // Resolve the promise with the response text
+          resolve(req.response);
+        } else {
+          // Otherwise reject with the status text
+          // which will hopefully be a meaningful error
+          if (window.ga) {
+            window.ga("send", "event", "serviceFailure", req.status, url);
+          }
+          notify(`Failed with status ${req.status}: ${req.statusText}`);
+          reject(Error(`Failed with status ${req.status}: ${req.statusText}`));
```

Comment:
If there's a reject, it looks like this function would continue to loop through it's list of URL after pushing the promise to the array. Is that correct?

If so, that could lead to missing chunks of data, which would be quite confusing.

Reply:
This seems to indicate promise.all will reject if any constituent promises reject: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

Reply:
unless the promises are being evaluated before promise.all is called, which is what I was trying to avoid.

Reply:
Add a test case for this and we should be good to go.